[KB-195] Methodology section (#123)
chhoumann/thesis-chemcam: report_thesis/src/sections/methodology.tex

\section{Methodology}\label{sec:methodology}
In this section, we outline the methodology used in this study to address the challenges identified in the problem definition. Our objective was to identify the most promising machine learning models and preprocessing techniques proposed from the literature, as outlined in Section~\ref{sec:related-work}, for predicting major oxide compositions from LIBS data.

> Then, using this knowledge, develop a pipeline that utilizes the strengths of these models and preprocessing techniques to improve prediction accuracy and robustness of the predictions.

Reviewer: Made task for this

We first describe the datasets used, including their preparation and the method of splitting for model training. Next, we outline the preprocessing steps and the model selection process, followed by a detailed explanation of the experimental setup and evaluation metrics. Finally, we discuss our validation testing procedures and the approach taken to ensure unbiased final model evaluations.

\subsection{Data Prepraration}
Similarly to our previous work \citet{p9_paper}, we used the publicly available \gls{ccs} data.
\gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis.

> The full description of this preprocessing can be found in \citet{wiensPreflightCalibrationInitial2013}.

Reviewer: ```suggestion
A comprehensive description of this preprocessing procedure is available in \citet{wiensPreflightCalibrationInitial2013}.
```

\begin{table*}[h]
\centering
\begin{tabular}{llllllll}
\toprule
     wave &         shot1 &         shot2 &  $\cdots$ &        shot49 &       shot50  & median        & mean          \\
\midrule
240.81100 & 6.4026649e+15 & 4.0429349e+15 & $\cdots$  & 1.7922483e+15 & 1.7126615e+15 & 1.9892956e+15 & 1.7561699e+15 \\
240.86501 & 3.8557462e+12 & 2.2923680e+12 & $\cdots$  & 1.1355429e+12 & 8.6930379e+11 & 7.8172542e+11 & 7.2805052e+11 \\
$\vdots$  & $\vdots$      & $\vdots$      & $\cdots$  & $\vdots$      & $\vdots$      & $\vdots$      & $\vdots$      \\
905.38062 & 1.8823427e+08 & 58500403.     & $\cdots$  & -8449286.6    & 8710775.0     & 4.0513312e+09 & 5.2188327e+09 \\
905.57349 & 1.9864713e+10 & 1.2956832e+10 & $\cdots$  & 1.9785415e+10 & 7.1994239e+09 & 1.1311150e+10 & 1.2201224e+10 \\
\bottomrule
\end{tabular}
\caption{Example of CCS data for a single target location (from \citet{p9_paper})}
\label{tab:ccs_data_example}

> \end{table*}

Reviewer 2: Can see we're using the label anywhere - need to reference this table?

Consequently, these regions are masked by zeroing out the values, rather than removing them, as they still provide valuable information for the model to learn from\cite{cleggRecalibrationMarsScience2017}.

Additionally, as a result of the aforementioned preprocessing applied to the raw \gls{libs} data, negative values are present in the \gls{ccs} data.
These negative values are not physically meaningful, since you cannot have negative light intensity.
Similar to the noisy edges, these negative values are also masked by zeroing out the values.

We transpose the data so that each row represents a target location and each column represents a wavelength feature.
Each location is now represented as a vector of wavelengths, with the corresponding average intensity values for each wavelength.
These vectors are then concatenated to form a matrix, where each row represents a target location and each column represents a wavelength feature.

For each target, we have a corresponding set of major oxide compositions in weight percentage (wt\%).
These compositions are used as the target labels for the machine learning models.

> An excerpt of this data is shown in Table \ref{tab:composition_data_example}.

Reviewer 2: Would it be easier for the reader to understand this part if we walked through the table? Just take one row and explain each column? Saying "The target, spectrum, and sample names are artifacts of the measurement progress. We primarily regard the sample name. The eight oxides are SiO2, ..., and serve as our ground truth values. The MOC total column is unused in our analysis" or something?

    0        & $\cdots$     & 5.09e+10 & 3.41e+10 & $\cdots$ & 0        & 57.60     & 0.78 & 26.60 & 2.73 & 0.70 & 0.01 & 0.38 & 7.10 & pg7         & pg7\_2013\_11\_07\_162038\_ccs \\
    0        & $\cdots$     & 5.99e+10 & 3.97e+10 & $\cdots$ & 0        & 57.60     & 0.78 & 26.60 & 2.73 & 0.70 & 0.01 & 0.38 & 7.10 & pg7         & pg7\_2013\_11\_07\_161422\_ccs \\
    0        & $\cdots$     & 5.22e+10 & 3.47e+10 & $\cdots$ & 0        & 57.60     & 0.78 & 26.60 & 2.73 & 0.70 & 0.01 & 0.38 & 7.10 & pg7         & pg7\_2013\_11\_07\_161735\_ccs \\
    0        & $\cdots$     & 5.29e+10 & 3.62e+10 & $\cdots$ & 0        & 57.60     & 0.78 & 26.60 & 2.73 & 0.70 & 0.01 & 0.38 & 7.10 & pg7         & pg7\_2013\_11\_07\_161552\_ccs \\
	$\vdots$ & $\cdots$ & $\vdots$ & $\vdots$ & $\cdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ & $\vdots$ \\
\midrule
\end{tabular}
\caption{Excerpt from the final dataset (values have been rounded to two decimal places for brevity).}
\label{tab:final_dataset_example}
\end{table*}

> \subsection{Model and Preprocessing Selection}

Reviewer 2: I think, particularly in this subsection, there's a lack of structure and depth. Most of the text _feels_ surface level, and it's kind of all-over-the-place.

It somewhat tracks the story, but skips over a lot of important details regarding the rationale behind our decisions.
And it completely omits the last month and a half of work (optimization framework, for example).

I think we need a lot more depth to make it work, as well as a well-formed structure. We probably need to spell out some outline for what exactly should be said here, as I don't think it's going to be a quick fix.

We did a ton of awesome work this semester, and I'm greedy enough on our behalf to want us to get credit for every single thing we did. Let's not downplay it by being brief.

Reviewer: Made task for this

> For the initial investigative experiments, we selected a range of models for further exploration, namely \gls{svr}, \gls{gbr}, \gls{pls}, \gls{xgboost}, \gls{ngboost}, \gls{etr}, \gls{enet}, and \gls{sgd}.

Reviewer 2: Need to ref back to related works to explain why we picked these? I can see we're kinda covering it in the next few lines, but maybe be really precise as to why we picked these? Not easy, of course. It's a while ago we actually did that work.

Also, the SuperCam model selection paper isn't mentioned here, which I believe was the inspiration for a good few models.

For instance, \citet{andersonImprovedAccuracyQuantitative2017} discussed the use of multiple regression models, finding that different models excelled with specific oxides, which informed our model-specific approach.
\citet{song_DF-K-ELM} presented a hybrid model combining domain knowledge with machine learning, which inspired our interest in models that could offer both high performance and interpretability. \citet{rezaei_dimensionality_reduction} demonstrated the beneficial impact of dimensionality reduction techniques like \gls{pca}, which we considered essential for managing our high-dimensional \gls{libs} data.

All models demonstrated robust performance; however, \gls{gbr}, \gls{svr}, \gls{xgboost}, \gls{etr}, and \gls{pls} consistently excelled across all oxides.
Notably, we observed model-specific strengths for certain oxides --- \gls{svr} was particularly effective for \ce{SiO2}, while \gls{gbr} excelled with \ce{FeO_T}.
This differential performance prompted us to explore architectural frameworks that could systematically capitalize on the strengths of each model for specific oxides.

> A particularly exciting discovery from our literature review was a study on the stacking and chaining of normalization methods initially aimed at classification contexts.

Reviewer 2: ```suggestion
An interesting discovery from our literature review was a study on the stacking and chaining of normalization methods initially aimed at classification contexts.
```

\section{Methodology}\label{sec:methodology}
In this section, we outline the methodology used in this study to address the challenges identified in Section~\ref{sec:problem_definition}. Our objective was to identify the most promising machine learning models and preprocessing techniques proposed from the literature, as outlined in Section~\ref{sec:related-work}, for predicting major oxide compositions from \gls{libs} data.
Then, using this knowledge, develop a pipeline that utilizes the strengths of these models and preprocessing techniques to improve prediction accuracy and robustness of the predictions.
We first describe the datasets used, including their preparation and the method of splitting for model training. Next, we outline the preprocessing steps and the model selection process, followed by a detailed explanation of the experimental setup and evaluation metrics. Finally, we discuss our validation testing procedures and the approach taken to ensure unbiased final model evaluations.

\subsection{Data Preparation}
Similarly to our previous work \citet{p9_paper}, we used the publicly available \gls{ccs} data from NASA's \gls{pds}~\cite{PDSGeoscienceNode}.

> \gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis.

Reviewer: ```suggestion
\gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps, as described by \citet{wiensPreflightCalibrationInitial2013}, such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis.
```

Then, using this knowledge, develop a pipeline that utilizes the strengths of these models and preprocessing techniques to improve prediction accuracy and robustness of the predictions.
We first describe the datasets used, including their preparation and the method of splitting for model training. Next, we outline the preprocessing steps and the model selection process, followed by a detailed explanation of the experimental setup and evaluation metrics. Finally, we discuss our validation testing procedures and the approach taken to ensure unbiased final model evaluations.

\subsection{Data Prepraration}
Similarly to our previous work \cite{p9_paper}, we used the publicly available \gls{ccs} data. 
\gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis. 
The full description of this preprocessing can be found in \citet{wiensPreFlight3}.

> While the CCS data is in a more suitable form for quantitative analysis, it still requires further preprocessing. Specifically, the data are still noisy at the spectrometer edges, and this noise has to be masked out. The data also contains negative values, which are not physically meaningful and need to be zeroed out.

Author: ```suggestion
While the \gls{ccs} data is in a more suitable form for quantitative analysis, it still requires further preprocessing. Specifically, the data are still noisy at the spectrometer edges, and this noise has to be masked out. The data also contains negative values, which are not physically meaningful and need to be zeroed out.
```

\section{Methodology}\label{sec:methodology}

> In this section, we outline the methodology used in this study to address the challenges identified in the problem definition. Our objective was to identify the most promising machine learning models and preprocessing techniques proposed in the literature, outlined in Section~\ref{sec:related-work}, for predicting major oxide compositions from LIBS data. 

Author: ```suggestion
In this section, we outline the methodology used in this study to address the challenges identified in the problem definition. Our objective was to identify the most promising machine learning models and preprocessing techniques proposed from the literature, as outlined in Section~\ref{sec:related-work}, for predicting major oxide compositions from LIBS data. 
```

Then, using this knowledge, develop a pipeline that utilizes the strengths of these models and preprocessing techniques to improve prediction accuracy and robustness of the predictions.

> We first present a description of the datasets used, how it was prepared and how it was split for model training. Then the preprocessing and model selection process is described followed by the experimental setup and the evaluation metrics employed. Finally, we describe our validation testing procedures and how the final model evaluations were conducted to ensure unbiased results.

Author: ```suggestion
We first describe the datasets used, including their preparation and the method of splitting for model training. Next, we outline the preprocessing steps and the model selection process, followed by a detailed explanation of the experimental setup and evaluation metrics. Finally, we discuss our validation testing procedures and the approach taken to ensure unbiased final model evaluations.
```

\subsection{Data Prepraration}
Similarly to our previous work \cite{p9_paper}, we used the publicly available \gls{ccs} data. 

> CCS refers to LIBS data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis. 

Author: ```suggestion
\gls{ccs} refers to \gls{libs} data that has been through a series of preprocessing steps such as subtracting the ambient light background, noise removal and removing the electron continuum to derive data that is more suitable for quantitative analysis. 
```

The full description of this preprocessing can be found in \citet{wiensPreFlight3}.

> While the CCS data is in a more suitable form for quantitative analysis, it still requires further preprocessing to be useful. Specifically, the data will still have noise at the boundaries of the spectrometer ranges, which have to be masked out. The data will also contain negative values, which are not physically meaningful and have to be set to zero.

Author: ```suggestion
While the CCS data is in a more suitable form for quantitative analysis, it still requires further preprocessing to be useful. Specifically, the data are still noisy at the spectrometer edges, and this noise has to be masked out. The data also contains negative values, which are not physically meaningful and need to be zeroed out.
```

> For a more detailed description of the data and the preprocessing steps, we refer the reader to \citet{p9_paper}, since the data preparation follows the same procedures detailed in that work.

Author: ```suggestion
For a more detailed description of the data and the preprocessing steps, we refer the reader to \citet{p9_paper}, as data preparation follows the same procedure detailed in that work.
```

Once preprocessed, the data was conventionally split into training and testing sets. 

> However, as detailed in \citet{p9_paper} and \cite{cleggRecalibrationMarsScience2017}, care has to be taken in order to ensure that the training set is constructed such that it has sufficient samples containing extreme values.

Author: ```suggestion
As detailed in \citet{p9_paper} and \citet{cleggRecalibrationMarsScience2017}, it is essential to include samples with extreme element compositions in the training set to ensure the model can generalize effectively.
```

> To incorporate these extreme values, the training set was constructed by including the two highest and two lowest extreme samples from each oxide category.

Author: ```suggestion
To incorporate these extreme values, the training set was constructed to include the two highest and two lowest samples from each oxide category.
```

> Following the inclusion of extreme samples, we adjusted the size of the test split to accommodate the reduced pool of normal samples. 

Author: Think this needs to be phrased, unsure what exactly you mean by this.

This adjustment was calculated by determining the number of extreme samples, subtracting this from the total sample count, and recalculating the test size proportionally. 
This ensures that the proportion of the test set remains appropriate relative to the now smaller dataset. 
Subsequently, we conducted the train-test split, where the normal samples --- those not categorized as extreme --- were divided according to the newly adjusted test size, with the extreme samples included to ensure exposure to varied data points.

\subsection{Model and Preprocessing Selection}

> For the initial investigative experiments we selected a range of models for further exploration, namely Support Vector Regression (SVR), Gradient Boosting Regression (GBR), Partial Least Squares Regression (PLS), eXtreme Gradient Boosting (XGBoost), Natural Gradient Boosting (NGBoost), Extra Trees Regression (ETR), Elastic Net (ENet), and Stochastic Gradient Descent (SGD). 

Author: ```suggestion
For the initial investigative experiments, we selected a range of models for further exploration, namely \gls{svr}, \gls{gbr}, \gls{pls}, \gls{xgboost}, \gls{ngboost}, \gls{etr}, \gls{enet}, and \gls{sgd}.
```

> We also included regular neural network (NN) and convolutional neural networks (CNN) in this phase of experimentation. 

Author: ```suggestion
We also included regular \gls{ann} and \gls{cnn} in this phase of experimentation. 
```

> This selection was guided primarily by the literature review but also by exploratory intuition,
> aiming to discover potentially innovative applications and performances within our specific dataset.

Author: ```suggestion
This selection was guided primarily by the literature review but also by exploratory intuition, aiming to discover potentially innovative applications and performances within our specific dataset.
```

> Our literature review highlighted various approaches and their effectiveness in handling the challenges associated with predicting major oxide compositions from LIBS data. 

Author: ```suggestion
Our literature review highlighted various approaches and their effectiveness in handling the challenges associated with predicting major oxide compositions from \gls{libs} data. 
```

For instance, Anderson et al. discussed the use of multiple regression models, finding that different models excelled with specific oxides, which informed our model-specific approach. 

> Song et al. presented a hybrid model combining domain knowledge with machine learning, which inspired our interest in models that could offer both high performance and interpretability. Rezaei et al. demonstrated the beneficial impact of dimensionality reduction techniques like PCA, which we considered essential for managing our high-dimensional LIBS data.

Author: ```suggestion
\citet{song_DF-K-ELM} presented a hybrid model combining domain knowledge with machine learning, which inspired our interest in models that could offer both high performance and interpretability. \citet{rezaei_dimensionality_reduction} demonstrated the beneficial impact of dimensionality reduction techniques like \gls{pca}, which we considered essential for managing our high-dimensional \gls{libs} data.
```

All models demonstrated robust performance; however, \gls{gbr}, \gls{svr}, \gls{xgboost}, \gls{etr}, and \gls{pls} consistently excelled across all oxides. 

> Notably, we observed model-specific strengths for certain oxides—SVR was particularly effective for \ce{SiO2}, while \gls{gbr} excelled with \ce{FeO_T}. 

Author: ```suggestion
Notably, we observed model-specific strengths for certain oxides --- \gls{svr} was particularly effective for \ce{SiO2}, while \gls{gbr} excelled with \ce{FeO_T}. 
```

This differential performance prompted us to explore architectural frameworks that could systematically capitalize on the strengths of each model for specific oxides.

A particularly exciting discovery from our literature review was a study on the stacking and chaining of normalization methods initially aimed at classification contexts. 
Inspired by these ideas, we explored the possibility of improving model performance by optimizing the preprocessing chain for each model, per oxide, in order to determine which normalization methods would be most beneficial.
For this purpose, we considered the following preprocessing techniques: Min-Max Scaling, Standard Scaling, Robust Scaling, MaxAbs Scaling, Quantile Transformer and Power Transformer. 

> Additionally, we also considered dimensionality reduction techniques such as PCA and KernelPCA.

Author: ```suggestion
Additionally, we also considered dimensionality reduction techniques such as \gls{pca} and \gls{kernel-pca}.
```

Through experimentation, we identified which preprocessing techniques were most effective for each model and oxide. 
Realizing the potential of this, we decided to further investigate how the predictive accuracy and robustness could be improved by combining the strengths of these models and preprocessing techniques.
From our investigation, we determined that the stacking ensemble method seemed very promising and novel in the LIBS analysis field, leading us to select it as the primary method for our study.

> Stacking ensemble is analogous to the methodologies employed in the original MOC pipeline, which also tailored predictions for each oxide by blending outputs from the PLS-SM and ICA phases, variably weighting the influence of the ICA predictions depending on the oxide.

Reviewer: Needs cite

Author: ok 👍

> Stacking ensemble is analogous to the methodologies employed in the original MOC pipeline, which also tailored predictions for each oxide by blending outputs from the PLS-SM and ICA phases, variably weighting the influence of the ICA predictions depending on the oxide.

Author: ```suggestion
Stacking ensemble is analogous to the methodologies employed in the original \gls{moc} pipeline, which also tailored predictions for each oxide by blending outputs from the \gls{pls1-sm} and \gls{ica} phases, variably weighting the influence of the \gls{ica} predictions depending on the oxide.
```
